pwiz_tools/Skyline/TestFunctional/DetectionsPlotTest.cs

```diff
@@ -174,29 +179,29 @@ protected override void DoTest()
         private void AssertDataCorrect(DetectionsPlotPane pane, int refIndex, float qValue, bool record = false)
         {
             DetectionPlotData data = null;
-            WaitForConditionUI(() => (data = pane.CurrentData) != null 
-                                           && data.QValueCutoff == qValue
+            WaitForConditionUI(() => pane.CurrentData != null 
+                                           && pane.CurrentData.QValueCutoff == qValue
                                            && DetectionPlotData.GetDataCache().Status == DetectionPlotData.DetectionDataCache.CacheStatus.idle,
                 () => $"Retrieving data for qValue {qValue}, refIndex {refIndex} took too long.");
             WaitForGraphs();
-            Assert.IsTrue(data.IsValid);
+            WaitForCondition(() => pane.CurrentData.IsValid);

```

Comment:
This is the fix for the assertion failure that Nick tried to fix. The problem was that WaitForGraphs() doesn't guarantee that the data computation is complete. I've seen this in the RT regression tests as well.